media_softlet/linux/common/ddi/media_libva_interface_next.cpp

```diff
@@ -1824,13 +1824,14 @@ VAStatus MediaLibvaInterfaceNext::PutImage(
         }
 
         //Copy data from image to temp surferce
-        MOS_STATUS eStatus = MOS_SecureMemcpy(tempSurfData, vaimg->data_size, imageData, vaimg->data_size);
-        if (eStatus != MOS_STATUS_SUCCESS)
+        MOS_STATUS eStatus = MOS_STATUS_SUCCESS;
+        if (tempMediaSurface->data_size >= vaimg->data_size)
         {
-            DDI_ASSERTMESSAGE("Failed to copy image to surface buffer.");
-            MediaLibvaUtilNext::UnlockSurface(tempMediaSurface);
-            DestroySurfaces(ctx, &tempSurface, 1);
-            return VA_STATUS_ERROR_OPERATION_FAILED;
+            eStatus = MOS_SecureMemcpy(tempSurfData, tempMediaSurface->data_size, imageData, vaimg->data_size);
+        }
+        else
+        {
+            eStatus = MOS_SecureMemcpy(tempSurfData, tempMediaSurface->data_size, imageData, tempMediaSurface->data_size);
         }
```

Comment:
Better keeping the error status check here

Reply:
Oops, sorry, lousy copy and paste job from me. I've fixed it and updated the PR.